packages/flutter_tools/test/hot_test.dart

```diff
@@ -228,6 +229,40 @@ void main() {
       Artifacts: () => mockArtifacts,
       HotRunnerConfig: () => TestHotRunnerConfig(successfulSetup: true, computeDartDependencies: false),
     });
+
+    testUsingContext('shutdown hook called after signal', () async {
+      final MockDevice mockDevice = MockDevice();
+      when(mockDevice.supportsHotReload).thenReturn(true);
+      when(mockDevice.supportsHotRestart).thenReturn(true);
+      when(mockDevice.supportsStopApp).thenReturn(false);
+      final List<FlutterDevice> devices = <FlutterDevice>[
+        FlutterDevice(mockDevice, generator: residentCompiler, trackWidgetCreation: false)
+      ];
+      await HotRunner(devices).cleanupAfterSignal();
+      expect(shutdownTestingConfig.shutdownHookCalled, true);
+      // Reset the state for the next test
+      shutdownTestingConfig.shutdownHookCalled = false;
+    }, overrides: <Type, Generator> {
+      Artifacts: () => mockArtifacts,
+      HotRunnerConfig: () => shutdownTestingConfig,
+    });
+
+    testUsingContext('shutdown hook called after app stop', () async {
+      final MockDevice mockDevice = MockDevice();
+      when(mockDevice.supportsHotReload).thenReturn(true);
+      when(mockDevice.supportsHotRestart).thenReturn(true);
+      when(mockDevice.supportsStopApp).thenReturn(false);
+      final List<FlutterDevice> devices = <FlutterDevice>[
+        FlutterDevice(mockDevice, generator: residentCompiler, trackWidgetCreation: false)
+      ];
+      await HotRunner(devices).preStop();
+      expect(shutdownTestingConfig.shutdownHookCalled, true);
+      // Reset the state for the next test
```

Comment:
Instead of doing this in each test case, wrap these in a `group` and then use the `setUp` method to create a new instance each time:

```
group('runPreShutdownOperations test', () {
  TestHotRunnerConfig shutdownTestingConfig;

  setUp(() {
    shutdownTestingConfig = TestHotRunnerConfig(
      successfulSetup: true,
      computeDartDependencies: false,
    );
  });
});
```

Reply:
Done